src/Faithlife.Analyzers/UriToStringAnalyzer.cs

```diff
@@ -0,0 +1,53 @@
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Faithlife.Analyzers
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public sealed class UriToStringAnalyzer : DiagnosticAnalyzer
+	{
+		public override void Initialize(AnalysisContext context)
+		{
+			context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+			context.RegisterCompilationStartAction(compilationStartAnalysisContext =>
+			{
+				var uriType = compilationStartAnalysisContext.Compilation.GetTypeByMetadataName("System.Uri");
+				if (uriType is null)
+					return;
+
+				compilationStartAnalysisContext.RegisterSyntaxNodeAction(c => AnalyzeSyntax(c, uriType), SyntaxKind.InvocationExpression);
+			});
+		}
+
+		private static void AnalyzeSyntax(SyntaxNodeAnalysisContext context, INamedTypeSymbol uriType)
+		{
+			var invocation = (InvocationExpressionSyntax) context.Node;
+
+			var method = context.SemanticModel.GetSymbolInfo(invocation.Expression).Symbol as IMethodSymbol;
+			if (method?.Name != "ToString" || method.ContainingType != uriType)
+				return;
+
+			var memberAccess = (MemberAccessExpressionSyntax) invocation.Expression;
+
+			context.ReportDiagnostic(Diagnostic.Create(s_rule, memberAccess.Name.GetLocation()));
+		}
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(s_rule);
+
+		public const string DiagnosticId = "FL0013";
+
+		private static readonly DiagnosticDescriptor s_rule = new(
+			id: DiagnosticId,
+			title: "Uri ToString Usage",
+			messageFormat: "Uri MAY NOT use .ToString()",
```

Comment:
```suggestion
			messageFormat: "Do not use Uri.ToString()",
```

This language feels out-of-place with other Faithlife Analyzers (and C# build warnings, in general).

(And if we were going full RFC 2119—which I'm not recommending—it would be "MUST NOT".)

Reply:
Changed verbiage is def better. However, right now it is listed as a warning rather than an error. 

`defaultSeverity: DiagnosticSeverity.Warning,`

Should this instead be a `.Error`?

Reply:
Given my recent update to the blog post (https://faithlife.codes/blog/2010/08/uritostring_must_die/); warning seems fine.